src/Pin/index.js

```diff
@@ -73,34 +83,46 @@ export default {
     )
     pinDialog = null
   },
-  submit(pin) {
+  submit(pin, context) {
     return new Promise((resolve, reject) => {
       try {
-        submit(pin)
-          .then(resolve)
-          .catch(reject)
+        if (contextCheck(context) || contextItems.includes(context)) {
```

Comment:
I would move the check for valid context (`contextItems.includes(context)`) to the `contextCheck` function. That makes it a bit cleaner to read, plus it's more DRY (since this includes check exists also in `unlocked`).